python/ray/data/dataset.py

```diff
@@ -1871,7 +1888,12 @@ def batch_block(block: ObjectRef[Block]):
 
         # Yield any remainder batches.
         if batcher.has_any() and not drop_last:
-            yield format_batch(batcher.next_batch(), batch_format)
+            with self._stats.iter_format_batch_s.timer():
+                result = format_batch(batcher.next_batch(), batch_format)
+            with self._stats.iter_user_s.timer():
+                yield result
+
+        self._stats.iter_total_s.add(time.perf_counter() - time_start)
```

Comment:
Any reason for not using the `.timer()` context manager for this one? Seems like all of this could be under that context manager:
```python
with self._stats.iter_total_s.timer():
    def format_batch(batch: Block, format: str) -> BatchType:
        ...
```

Reply:
Ah it was too indented. Didn't seem worth the tradeoff!